### PR TITLE
5pm-2 kt - Make table of course search results invisible initially and if there are no search results

### LINKED ISSUE #273: In Search By Course Name make the table invisible initially
# User Story

As a user I want to NOT see the table header on the Search By Course Name before I have searched so that I am not confused about missing data.

# Discussion

When you first visit the page: https://proj-ucsb-courses-search.herokuapp.com/history/courseName

Even before you do a search, the table headers show up, as seen below:

image

What we want is that the headers don't show up unless there is some data.

# Acceptance Criteria

- [x] Table headers don't show up when there is no data
- [x] Table headers do show up when there is data.

# Implementation Todos

The React component code is in `/pages/History/CourseName`

The component that displays the table is: 

```
  <BasicCourseTable classes={courseJSON.classes} checks={[cancelled,closed,full]} displayQuarter allowExport={true}/>
```

If we make the change INSIDE THAT COMPONENT, then the change will be effective pages that use that component.  That's probably  a good idea.  

Here's the filename for `BasicCourseTable`: `javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js`

This code is what makes the table show up:

```
return (
    <ToolkitProvider
      keyField="uniqueKey"
      data={sections}
      columns={csvColumns}
      exportCSV={{
        fileName: 'CourseTable.csv',
        blobType: 'csv/text;charset=utf-8'
      }}
    >
      {
        props => (
          <div>
            <ExportCSVButton {...props.csvProps} />
            <hr />
            <BootstrapTable keyField="uniqueKey" columns={columns} rowStyle={rowStyle} {...props.baseProps} />
          </div>
        )
      }
    </ToolkitProvider>
  );
```

What if, above this return, we put in something like this:

```
if (sections===[] || sections===null) {
    return (
      <div data-testid="no-course-data">
      </div>
    )
  }
```

Then we should see nothing on the page when `sections` is empty or null, but we can write tests that check for the presence of that empty html `<div>` element by looking for it by it's "test id".

### PR DESCRIPTION
**Pull request by 320trankt**
- Tweaks BasicCourseTable.js to return an empty div element with id="no-course-data" if the search's resultant courses array is empty.
- Fixes #273

### CODE REVIEW

**Review by sydneylim — APPROVED** [review]
LGTM

**Review by mmunozvaltierra — APPROVED** [review]
LGTM

**Review by Christine-ong — DISMISSED** [review]
LGTM

**Review by sriya112 — COMMENTED** [review]
LGTM

**Review by sriya112 — APPROVED** [review]
LGTM

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 1]
```diff
@@ -200,6 +200,14 @@ const BasicCourseTable = ({ classes, checks, displayQuarter, allowExport }) => {
...
     }
   );
+
+  if ((sections.length === 0) || (sections===null)) {
+    return (
+      <div data-testid="no-course-data">
+      </div>
+    )
+  }
+
```
Comment (pconrad) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:210]: We should really make both a test for this case, and a storybook for this case.

**Review by pconrad — CHANGES_REQUESTED** [review]
I'd like to see a test for this, and a storybook story for this.

I'll walk the team through both of those.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js (hunk 1 above)
Comment (pconrad) [reply javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:210]: Oh, nvm about the story part... that's already done!  Let's just focus on the testing part.

**Review by pconrad — APPROVED** [review]
LGTM!  testing now addressed

**Review by sriya112 — APPROVED** [review]
LGTM

**Pull request merged**